### PR TITLE
LPD-8012 Send user notifications when publication fails to publish

### PR DESCRIPTION
**Pull request by brookedalton**
https://liferay.atlassian.net/browse/LPD-8012

### CODE REVIEW

File: modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/internal/background/task/CTPublishBackgroundTaskExecutor.java [hunk 1]
```diff
@@ -100,7 +108,7 @@ public BackgroundTaskResult execute(BackgroundTask backgroundTask)
 		if (!_ctSchemaVersionLocalService.isLatestCTSchemaVersion(
 				fromCTCollection.getSchemaVersionId())) {
 
-			throw new IllegalArgumentException(
+			throw new CTPublishConflictException(
```
Comment (pei-jung) [inline modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/internal/background/task/CTPublishBackgroundTaskExecutor.java:112]: Don't change the exception type. This exception happens the publication is out of date, meaning there's a db schema change.

File: modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/internal/background/task/CTPublishBackgroundTaskExecutor.java [hunk 2]
```diff
@@ -235,6 +243,39 @@ public BackgroundTaskDisplay getBackgroundTaskDisplay(
 		return new CTPublishBackgroundTaskDisplay(backgroundTask);
 	}
 
+	@Override
+	public String handleException(
+		BackgroundTask backgroundTask, Exception exception) {
+
+		long fromCTCollectionId = MapUtil.getLong(
```
Comment (pei-jung) [inline modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/internal/background/task/CTPublishBackgroundTaskExecutor.java]: Move var declaration close to its usage.

File: modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/internal/background/task/CTPublishBackgroundTaskExecutor.java [hunk 3]
```diff
@@ -247,30 +248,33 @@ public BackgroundTaskDisplay getBackgroundTaskDisplay(
 	public String handleException(
 		BackgroundTask backgroundTask, Exception exception) {
 
-		long fromCTCollectionId = MapUtil.getLong(
-			backgroundTask.getTaskContextMap(), "fromCTCollectionId");
+		if (FeatureFlagManagerUtil.isEnabled("LPD-8012")) {
```
Comment (pei-jung) [inline modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/internal/background/task/CTPublishBackgroundTaskExecutor.java:251]: Do an early return if feature flag is not enabled so we can keep the code flat, eliminating the extra indentation.

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 4]
```diff
@@ -7252,6 +7252,8 @@ feature.flag.COMMERCE-12754.description=Users can issue refunds to an order's or
 feature.flag.COMMERCE-12754.title=Refund Order Payments
 feature.flag.COMMERCE-13024.description=Users can view and edit custom object fields for users and accounts via the Organization Management widget.
 feature.flag.COMMERCE-13024.title=Organization Management Widget and Custom Object Fields
+feature.flag.LPD-8012.description=Send customized Publication information to users.
+feature.flag.LPD-8012.title=Register Publication notifications
```
Comment (pei-jung) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:7256]: Leave out the FF title and description for now. We'll add it when Laszlo have these defined.

File: portal-impl/src/portal.properties [hunk 5]
```diff
@@ -6373,6 +6373,11 @@
     #
     feature.flag.COMMERCE-13024=false
 
+    #
+    # Env: LIFERAY_FEATURE_PERIOD_FLAG_PERIOD__UPPERCASEL__UPPERCASEP__UPPERCASED__MINUS__NUMBER8__NUMBER0__NUMBER1__NUMBER2_
+    #
+    feature.flag.LPD-8012=false
```
Comment (pei-jung) [inline portal-impl/src/portal.properties:6379]: We should use the epic LPD-11018 as the FF.

**Review by pei-jung — CHANGES_REQUESTED** [review]

**Comment by brookedalton** [comment]
Resent in https://github.com/liferay-publications/liferay-portal/pull/986 due to wrong ticket name

**Pull request closed without merging**